juntagrico/views_subscription.py

```diff
@@ -446,42 +440,38 @@ def change_nickname(request, subscription_id):
 
 
 @login_required
-def order_shares(request):
+def manage_shares(request):
     if request.method == 'POST':
-        referer = request.POST.get('referer')
         try:
             shares = int(request.POST.get('shares'))
             shareerror = shares < 1
         except ValueError:
             shareerror = True
         if not shareerror:
             create_share(request.user.member, shares)
-            return redirect('{}?referer={}'.format(reverse('share-order-success'), referer))
+            return redirect(reverse('manage-shares'))
     else:
         shareerror = False
-        if request.META.get('HTTP_REFERER')is not None:
-            referer = request.META.get('HTTP_REFERER')
-        else:
-            referer = 'http://' + Config.adminportal_server_url()
+    member = request.user.member
+    shares = member.share_set
     renderdict = get_menu_dict(request)
+    overflow_list = [shares.count()]
+    overflow_list.append(member.subscription_future.share_overflow) if member.subscription_future is not None else 0
```

Comment:
If a share is cancelled, it won't count towards the overflow anymore and the list will show: "cancelled - This share is required for a sub"
instead of counting the overflow it would be easier to count the `required_shares` for current and future sub and take the the max there.

Reply:
you are right in the analysis of the error but the solution does not work
asume A B and C are member which share a subscription which requires 2 shares. Each member has 1 share.
So one of them could cancel his share and they would still have enough shares for the subscription.
with your solution no one if them could cancel their share.
the solution is to use the usable shares for the calculation

Reply:
You're right. I commented it for the next generations.